src/controllers/routine/get-lectures-by-teacher-id.ts

```diff
@@ -0,0 +1,71 @@
+import { PrismaClient } from '@prisma/client';
+
+import asyncHandler from '../../utils/asyncHandler';
+
+const { lecture } = new PrismaClient();
+
+/**
+ * @auth required
+ * @route {POST} /routine/get_routine_by_teacher_id
+ * @body {week: boolean, day: string}
+ * @returns {teachers routine}
+ */
+
+const getLectureByTeacherId = asyncHandler(async (req, res) => {
+  try {
+    const { day } = req.query as unknown as {
+      day: string;
+    };
+
+    const { id, schoolId, type } = req;
+
+    let teacherId = id;
+    type
+      ? 'PRINCIPAL' || 'SENIOR_TEACHER'
+      : (teacherId = req.query.teacherId as string);
```

Comment:
@Prodip-Kumar-Paul please write down a comment for better understanding

Reply:
will write now onwards